ruler/configs.go

```diff
@@ -0,0 +1,111 @@
+package ruler
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/rules"
+)
+
+// TODO: Extract configs client logic into go client library (ala users)
+
+type configID int
+
+type cortexConfig struct {
+	RulesFiles map[string]string `json:"rules_files"`
```

Comment:
Would help readability to have a comment "Map from rule file name to content." explaining the `map[string]string` here.